buildpacks/gradle/src/errors.rs

```diff
@@ -0,0 +1,98 @@
+            ", error = error },
+        ),
+        GradleBuildpackError::GradleBuildUnexpectedStatusError(exit_status) => {
+            let exit_code_string = exit_status.code().map_or_else(
+                || String::from("<unknown>"),
+                |exit_code| exit_code.to_string(),
+            );
+
+            log_error(
+                "Failed to build app with Gradle",
+                formatdoc! {"
+                    We're sorry this build is failing! If you can't find the issue in application code,
+                    please submit a ticket so we can help: https://help.heroku.com/
+
+                    Gradle exit code was: {exit_code}
+                ", exit_code = exit_code_string },
+            );
+        }
+        GradleBuildpackError::GetTasksError(error) => log_please_try_again_error(
+            "Failed to get Gradle tasks",
+            "Failed to get Gradle tasks",
+            error,
+        ),
+        GradleBuildpackError::GetDependencyReportError(error) => log_please_try_again_error(
+            "Failed to get Gradle dependency report",
+            "Failed to get Gradle dependency report",
+            error,
+        ),
+        GradleBuildpackError::WriteGradlePropertiesError(error) => log_please_try_again_error(
+            "Failed to write Gradle configuration",
+            "Failed to write Gradle configuration",
+            error,
+        ),
+        GradleBuildpackError::WriteGradleInitScriptError(error) => log_please_try_again_error(
+            "Failed to write Gradle init script",
+            "Failed to write Gradle init script",
+            error,
+        ),
+        GradleBuildpackError::CannotSetGradleWrapperExecutableBit(error) => {
+            log_please_try_again_error(
+                "Failed to set executable bit for Gradle wrapper",
+                "Failed to set executable bit for Gradle wrapper",
+                error,
+            );
+        }
+        GradleBuildpackError::StartGradleDaemonError(error) => log_error(
+            "Failed to start Gradle daemon",
+            formatdoc! {"
+                We're sorry this build is failing! If you can't find the issue in application code,
+                please submit a ticket so we can help: https://help.heroku.com/
+
+                Details: {error:?}
+            ", error = error },
+        ),
+        GradleBuildpackError::BuildTaskUnknown => log_error(
+            "Failed to determine build task",
+            indoc! {"
+                It looks like your project does not contain a 'stage' task, which Heroku needs in order
+                to build your app. Our Dev Center article on preparing a Gradle application for Heroku
+                describes how to create this task:
+                https://devcenter.heroku.com/articles/deploying-gradle-apps-on-heroku
+
+                If you're stilling having trouble, please submit a ticket so we can help:
+                https://help.heroku.com
```

Comment:
This seems an awful lot like "open a ticket and we'll debug your application code (even though this is fully documented and tested)" which is outside of scope of support?

Reply:
Yeah, the messages need overhaul across all CNBs. This is the message we have in the classic buildpack currently.